### DISCUSSION TITLE
multiple minutes of startup time for obsidian using metadata menu

### DISCUSSION THREAD

**Discussion by ben1mit** [mdelobelle/metadatamenu#263] [Q&A]
When I have metadata menu enabled obsidian takes 3-4 minutes of starting up, and the app looks like this: 

<img width="619" alt="Skjermbilde 2022-12-31 kl  13 59 04" src="https://user-images.githubusercontent.com/48367359/210138572-4fe8f3fc-0b05-4b08-ab58-719bfdd291f8.png">

I am using obsidian 1.1.9, and metadata menu version 0.4.4 on a mac.

when I disable metadata menu the vault takes only a couple of seconds to load. 
I have a relatively large vault ( a couple of thousand notes) that takes a while to index, so this might be a contributing factor. 
the bug started to appear after the update that introduces canvas fields.
I attempted to move the folder containing the metadata menu file classes out of the vault, which made no difference to the start up time. 
I also tried resetting the fileClass field alias back to the default (which means that none  of my notes have the file class field). Doing this brought the startup time down to about 2 minutes

"Metadata Menu loaded" shows up within a couple  of seconds in the console. 
No warnings appear in the console, it simply takes a long while  for the app to start.

**Comment by mdelobelle** [comment]
Hi do you use lookup or formulas fields in your vault?
If yes could you try to disable the "auto update" and see if it improves the speed?

**Comment by ben1mit** [comment]
I'm not using lookup fields or formulas fields

**Comment by ben1mit** [comment]
It  may  be that this issue is not actually a problem with metadata menu, because now I am having problems with my startup time regardless of having metadata menu disabled. 

I tested multiple times earlier, and it seemed that every time I disabled metadata menu the app loaded fast and with it enabled it took several minutes, but now this seems not to be the case. 

I will look further into it and see if I can figure out what the actual issue is

**Comment by mdelobelle** [comment]
Ok! looking forward to hearing from you about that

**Comment by mdelobelle** [comment]
Hi any news about this issue?

**Comment by PhyberApex** [comment]
Sorry for hijacking this but I had exactly this problem and a auto update disable on my fileClass field did the trick. But now I wonder how ofter is that field refreshed now that I disabled auto-update? Once per opening of the vault? Once per app start? Never?

~Cheers

**Comment by mdelobelle** [comment]
> Hi any news about this issue?

@ben1mit are you still experiencing slowness from metadata-menu?

**Reply by ben1mit** [reply]
I am still experiencing slowness, but I have been unable to pinpoint any one plugin that causes it, so it  might have nothing to do with metadata menu.
The problem seems to come from the combination of several plugins, so my current workaround is to just edit the community-plugins.json and remove a little chunk of plugins when I turn on obsidian, and then paste it back in when obsidian has finished starting (which then only takes a few seconds)

**Reply by ben1mit** [reply]
I found this repository just now: https://github.com/ohm-en/obsidian-plugin-manager?ref=obsidian-iceberg
it might solve my problem. 
My current workaround  works fine for now, but I'll test the plugin manager once it is gone  through review and appears in the community plugins list

**Comment by mdelobelle** [comment]
> Sorry for hijacking this but I had exactly this problem and a auto update disable on my fileClass field did the trick. But now I wonder how ofter is that field refreshed now that I disabled auto-update? Once per opening of the vault? Once per app start? Never?
> 
> ~Cheers

You'll have to trigger it manually through commands in the command palette:
<img width="541" alt="image" src="https://user-images.githubusercontent.com/1233240/213861661-5248b2de-952b-44e4-beed-8a9e0ce3f006.png">

You can set hotkeys for them

**Reply by PhyberApex** [reply]
Hey thanks for the answer. What ever you did with your latest update. It fixed the slowness for me completly! Btw the null values are gone and the autocomplete is working MUCH faster! Awesome work man!
